web_src/js/utils/dom.js

```diff
@@ -0,0 +1,65 @@
+function getComputedStyleProperty(el, prop) {
+  const cs = el ? window.getComputedStyle(el) : null;
+  return cs ? cs[prop] : null;
+}
+
+function isShown(el) {
+  return getComputedStyleProperty(el, 'display') !== 'none';
+}
+
+function assertShown(el, expectShown) {
+  if (window.config.runModeIsProd) return;
```

Comment:
Better to move this outside for a tiny perf gain as JS does not have to call the function every time.

Reply:
Moving it out will make the caller's code quite complex and dirty.

And see the comment:

> this assertion can be removed after next release cycle or if it has been proved that there is no bug.